src/FSharp.Core/collections.fs

```diff
@@ -8,41 +8,43 @@ open Microsoft.FSharp.Core
 open Microsoft.FSharp.Core.Operators
 open System.Collections.Generic
 
-module HashIdentity = 
-            
-    let inline Structural<'T when 'T : equality> : IEqualityComparer<'T> = 
+module HashIdentity =
+
+    let inline Structural<'T when 'T: equality> : IEqualityComparer<'T> =
         LanguagePrimitives.FastGenericEqualityComparer<'T>
-          
-    let inline LimitedStructural<'T when 'T : equality>(limit) : IEqualityComparer<'T> = 
+
+    let inline LimitedStructural<'T when 'T: equality> (limit) : IEqualityComparer<'T> =
         LanguagePrimitives.FastLimitedGenericEqualityComparer<'T>(limit)
-          
-    let Reference<'T when 'T : not struct > : IEqualityComparer<'T> = 
+
+    let Reference<'T when 'T: not struct> : IEqualityComparer<'T> =
         { new IEqualityComparer<'T> with
-              member _.GetHashCode(x) = LanguagePrimitives.PhysicalHash(x) 
-              member _.Equals(x,y) = LanguagePrimitives.PhysicalEquality x y }
+            member _.GetHashCode(x) = LanguagePrimitives.PhysicalHash(x)
+            member _.Equals(x, y) = LanguagePrimitives.PhysicalEquality x y }
 
-    let inline NonStructural< 'T when 'T : equality and 'T  : (static member ( = ) : 'T * 'T    -> bool) > = 
+    let inline NonStructural<'T when 'T: equality and 'T: (static member (=): 'T * 'T -> bool)> =
         { new IEqualityComparer<'T> with
-              member _.GetHashCode(x) = NonStructuralComparison.hash x 
-              member _.Equals(x, y) = NonStructuralComparison.(=) x y  }
+            member _.GetHashCode(x) = NonStructuralComparison.hash x
+            member _.Equals(x, y) = NonStructuralComparison.(=) x y }
+
+    let inline FromFunctions hasher equality : IEqualityComparer<'T> =
+        let eq = OptimizedClosures.FSharpFunc<_, _, _>.Adapt (equality)
 
-    let inline FromFunctions hasher equality : IEqualityComparer<'T> = 
-        let eq = OptimizedClosures.FSharpFunc<_,_,_>.Adapt(equality)
-        { new IEqualityComparer<'T> with 
+        { new IEqualityComparer<'T> with
             member _.GetHashCode(x) = hasher x
-            member _.Equals(x,y) = eq.Invoke(x,y)  }
+            member _.Equals(x, y) = eq.Invoke(x, y) }
 
-module ComparisonIdentity = 
+module ComparisonIdentity =
 
-    let inline Structural<'T when 'T : comparison > : IComparer<'T> = 
+    let inline Structural<'T when 'T: comparison> : IComparer<'T> =
         LanguagePrimitives.FastGenericComparer<'T>
 
-    let inline NonStructural< 'T when 'T : (static member ( < ) : 'T * 'T    -> bool) and 'T : (static member ( > ) : 'T * 'T    -> bool) > : IComparer<'T> = 
+    let inline NonStructural<'T when 'T: (static member (<): 'T * 'T -> bool) and 'T: (static member (>):
+        'T * 'T -> bool)> : IComparer<'T> =
```

Comment:
Should line breaks instead be aligned with `and` and only then `:`?

Reply:
Yes this looks a little dodgy.  @nojaf similar question to above - will constraint formatting affect this? thanks

Reply:
No, reported https://github.com/fsprojects/fantomas/issues/2267.